### PR TITLE
members search,view and update features added

### PR DESCRIPTION
**Pull request by Mouroof**

### CODE REVIEW

File: application/rest/beneficiary.py [hunk 1]
```diff
@@ -54,4 +55,37 @@ def create_beneficiary():
...
+        json.dumps(response.value, cls=BeneficiaryJsonEncoder),
+        mimetype="application/json",
+        status=STATUS_CODES[response.type],
+    )
+
+@blueprint.route("/api/v1/beneficiaries/<id>", methods=["POST"])
+def beneficiary_update(id):
+    firstName = request.json['firstName']
+    lastName = request.json['lastName']
+    middleName = request.json['middleName']
```
Comment (ik-nitk) [inline application/rest/beneficiary.py:81]: We need govt. id number for beneficiaries

File: application/rest/beneficiary.py [hunk 2]
```diff
@@ -54,4 +55,37 @@ def create_beneficiary():
...
+    response = view_beneficiary(current_app.config.get('REPO'), id)
+    return Response(
+        json.dumps(response.value, cls=BeneficiaryJsonEncoder),
+        mimetype="application/json",
+        status=STATUS_CODES[response.type],
+    )
+
+@blueprint.route("/api/v1/beneficiaries/<id>", methods=["POST"])
+def beneficiary_update(id):
+    firstName = request.json['firstName']
```
Comment (ik-nitk) [inline application/rest/beneficiary.py:79]: hope you are removing whitespaces around names.

File: application/rest/beneficiary.py (hunk 1 above)
Comment (Mouroof) [reply application/rest/beneficiary.py:81]: In beneficiary domain we dont have govt_id. should I add govt_id in domain.

File: application/rest/beneficiary.py (hunk 2 above)
Comment (Mouroof) [reply application/rest/beneficiary.py:79]: No.I didn't remove whitespaces

Comment (ik-nitk) [reply application/rest/beneficiary.py:79]: Next time add these checks and validations on server .  Its always good to be resilient programmer

File: cms/repository/postgresrepo.py [hunk 3]
```diff
@@ -174,3 +174,102 @@ def create_case(self, title,purpose,description, contact_details, contact_addres
         session.commit()
         return new_case.case_id
 
+    def search_member(self, search_input):
```
Comment (ik-nitk) [inline cms/repository/postgresrepo.py:177]: This file starts growing . we should organise into small files like we did for domain use cases

File: web/backend_api_builder.py [hunk 4]
```diff
@@ -22,6 +22,10 @@ def member_id(self, member_id):
         """
         return self.members + '/{}'.format(member_id)
 
+    @property
+    def member_search(self):
+        return self.base + 'members/search'
```
Comment (ik-nitk) [inline web/backend_api_builder.py:27]: you can just do `self.members` + '/search'

File: web/member/views.py [hunk 5]
```diff
@@ -35,10 +35,69 @@ def create_member():
...
+        url = api.member_search
+        response = session.post(url, json = {"search_input":search_input})
+        response.raise_for_status()
+        members = response.json()
+        return jsonify({'htmlresponse': render_template("members/search_response.html", members=members)})
+
+
+@blueprint.route("/members/view/<id>",methods = ["GET"])  
+def member_view(id):
+    print(id)
```
Comment (ik-nitk) [inline web/member/views.py:62]: remove all the prints from the code

File: web/member/views.py [hunk 6]
```diff
@@ -35,10 +35,69 @@ def create_member():
...
+    api = current_app.config.get('api')
+    session = current_app.config.get('session') 
+    url = api.member_id(id)
+    print(url)
+    response = session.get(url)
+    response.raise_for_status()
+    members = response.json()
+    print(members)
```
Comment (ik-nitk) [inline web/member/views.py:70]: use logs

**Review by ik-nitk — APPROVED** [review]

File: web/backend_api_builder.py (hunk 4 above)
Comment (Mouroof) [reply web/backend_api_builder.py:27]: ok

**Pull request merged**